### PR TITLE
fix(plan): rewrite `node_modules/.bin/*.cmd` to powershell on Windows

### LINKED ISSUE #1176: Pressing Ctrl+C after vp run dev, pnpm run dev, or npm run dev leaves the terminal in a broken state on Windows
### Describe the bug

 - On Windows, after starting the dev server and stopping it with Ctrl+C, the terminal is left in a broken or unusable state.
 - Common symptoms include:
   - terminal input becomes sluggish or freezes
   - Backspace behaves incorrectly and prints ^H 
   - Ctrl+C is echoed as ^C
 - This issue is reproducible with `vp run dev`, `pnpm run dev`, and `npm run dev`.

- The issue does not reproduce with:
   - `bun run dev`
   - `.\node_modules\.bin\vite.CMD`
   - `.\node_modules\.bin\vite.ps1`
   - `C:\Users\Curtion\Desktop\node-v24.14.1-win-x64\npm.cmd run dev`(Native Node.js)
   - `C:\Users\Curtion\Desktop\node-v24.14.1-win-x64\npm.ps1 run dev`(Native Node.js)

https://github.com/user-attachments/assets/275ac35b-f031-45e3-ae7d-29261af21c4a

### Reproduction

https://github.com/Curtion/report-vite-plus-1

### Steps to reproduce

 - Open a terminal on Windows.
 - Go to the project root directory.
 - Run one of the following commands:
   - `vp run dev`
   - `pnpm run dev`
   - `npm run dev`
 - Wait for the dev server to start.
 - Press Ctrl+C once to stop the process.
 - Observe that the terminal enters an abnormal state:
   - input becomes slow or unresponsive
   - Backspace prints ^H
   - Ctrl+C prints ^C

### System Info

```shell
@Curtion ➜ report-vite-plus-1 git(main) vp env current
VITE+ - The Unified Toolchain for the Web

Environment:
  Version  24.14.1
  Source   lts

Tool Paths:
  node  C:\Users\Curtion\.vite-plus\js_runtime\node\24.14.1\node.exe
  npm   C:\Users\Curtion\.vite-plus\js_runtime\node\24.14.1\npm.cmd
  npx   C:\Users\Curtion\.vite-plus\js_runtime\node\24.14.1\npx.cmd
@Curtion ➜ report-vite-plus-1 git(main) vp --version  
VITE+ - The Unified Toolchain for the Web

vp v0.1.14

Local vite-plus:
  vite-plus  Not found

Tools:
  vite             Not found
  rolldown         Not found
  vitest           Not found
  oxfmt            Not found
  oxlint           Not found
  oxlint-tsgolint  Not found
  tsdown           Not found

Environment:
  Package manager  pnpm v10.28.0
  Node.js          v24.14.1
@Curtion ➜ report-vite-plus-1 git(main)
```

### Used Package Manager

pnpm

### Logs

```shell

```

### Validations

- [x] Read the [Contributing Guidelines](https://github.com/voidzero-dev/vite-plus/blob/main/CONTRIBUTING.md).
- [x] Check that there isn't [already an issue](https://github.com/voidzero-dev/vite-plus/issues) for the same bug.
- [x] Confirm this is a Vite+ issue and not an upstream issue (Vite, Vitest, tsdown, Rolldown, or Oxc).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example).

### PR DESCRIPTION
**Pull request by fengmk2**
## Summary

Running a `node_modules/.bin/*.cmd` shim on Windows triggers `cmd.exe`'s *"Terminate batch job (Y/N)?"* prompt on Ctrl+C, which corrupts the terminal. At plan time, when `which()` resolves such a shim inside the workspace and a sibling `.ps1` exists, rewrite the invocation to run the `.ps1` directly through PowerShell — the spawn layer stays untouched and Ctrl+C propagates cleanly.

Closes voidzero-dev/vite-plus#1176

## What the rewrite does

```
program_path: <abs path to pwsh.exe or powershell.exe>
args:         [-NoProfile, -NoLogo, -ExecutionPolicy, Bypass, -File,
               <.ps1 path, relative to task cwd>, ...original_args]
```

The `.ps1` path is stored **relative to the task's cwd** (with `/` separators), so `SpawnFingerprint.args` is portable across machines — no absolute paths leak into cache keys. PowerShell resolves `-File <relative>` against its own working directory (the task's cwd) and lands on the correct file.

## When it fires

Only when **all** of the following hold:

1. Extension is `.cmd` (case-insensitive) — cmd-shim never emits `.bat`.
2. Path lives **inside the workspace root** — global shims like `%AppData%\npm\node_modules\.bin\foo.cmd` are left alone.
3. Path's last two components are `.bin` / `node_modules` (case-insensitive).
4. A sibling `.ps1` exists.
5. `pwsh.exe` or `powershell.exe` is on PATH.

Any miss and the original `.cmd` is kept — behavior matches pre-PR.

## Tests

- **6 cross-platform unit tests** in `vite_task_plan::ps1_shim::tests` cover the pure rewrite logic: workspace root, hoisted monorepo subpackage (`..` traversal), missing sibling, non-`.cmd` extensions, `.cmd` outside `.bin`, `.cmd` outside the workspace.
- **Windows-only plan snapshot** in `plan_snapshots/fixtures/windows_cmd_shim_rewrite/` — a pnpm workspace with a sub-package. Two plan cases (`dev_in_subpackage` via `cwd`, `dev_filter_from_root` via `--filter`) produce **byte-identical** snapshots, proving the cache key doesn't depend on how the user navigates to the task. The harness gained a `windows_only: bool` flag on `SnapshotsFile`; `redact_snapshot` gained a pass that maps `pwsh` / `powershell` and their absolute host paths to `<powershell>` so the snapshot doesn't pin a specific runner image.

## Test plan

- [x] `cargo clippy --workspace --all-targets` + `cargo test --workspace` green on macOS
- [x] Windows CI green
- [ ] Manual on Windows: https://github.com/Curtion/report-vite-plus-1, Ctrl+C during `vp run dev` leaves terminal clean
- [ ] Manual: confirm `.cmd` fallback on a Windows box with PATH stripped of PowerShell
- [ ] Manual: confirm a globally installed `.cmd` outside the workspace is *not* rewritten

---

> [!NOTE]
> **Medium Risk**
> Changes Windows command planning and spawn fingerprints by rewriting certain resolved executables and arguments, which could affect task execution and cache keys on that platform. Scope is constrained to `.cmd` shims under workspace `node_modules/.bin` with a `.ps1` sibling and a detectable PowerShell host.
> 
> **Overview**
> On Windows, planning now rewrites resolved `node_modules/.bin/*.cmd` shims inside the workspace to run the sibling `.ps1` via `pwsh.exe`/`powershell.exe` (adding a fixed PowerShell arg prefix and recording the `.ps1` path *relative to the task cwd*), so Ctrl+C no longer triggers `cmd.exe`'s "Terminate batch job" prompt and cache fingerprints reflect the actual invocation.
> 
> This adds a dedicated `ps1_shim` module (using `pathdiff`) and applies it to both normal spawn commands and synthetic plan requests. Snapshot test infrastructure is updated with per-fixture `platform` filtering plus PowerShell path/name redaction, and a Windows-only plan snapshot fixture is added to validate stable, portable cache keys across invocation styles.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 09b4dc1b88294e8851c449995d7b20a2cba57151. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by fengmk2** [comment]
@cursor review

**Comment by fengmk2** [comment]
@cursor review

**Comment by fengmk2** [comment]
@cursor review

**Comment by fengmk2** [comment]
before

image

after

image

**Comment by fengmk2** [comment]
@cursor review

**Comment by fengmk2** [comment]
@cursor review

File: crates/vite_task_plan/src/ps1_shim.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+#[cfg(not(windows))]
+pub const fn prefer_ps1_sibling(resolved: AbsolutePathBuf) -> AbsolutePathBuf {
+    resolved
+}
+
+#[cfg(any(windows, test))]
+fn find_ps1_sibling(resolved: &vite_path::AbsolutePath) -> Option<AbsolutePathBuf> {
+    let path = resolved.as_path();
+    let ext = path.extension().and_then(|e| e.to_str())?;
+    if !(ext.eq_ignore_ascii_case("cmd") || ext.eq_ignore_ascii_case("bat")) {
```
Comment (fengmk2) [inline crates/vite_task_plan/src/ps1_shim.rs]: Only need to check `cmd`, no need to process the `bat`

**Comment by fengmk2** [comment]
@cursor review

**Comment by fengmk2** [comment]
@cursor review

**Comment by fengmk2** [comment]
@cursor review

**Comment by fengmk2** [comment]
@cursor review

**Comment by fengmk2** [comment]
@cursor review

File: crates/vite_task_plan/src/ps1_shim.rs [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+/// Compute a path that, when joined onto `from`, reaches `to`. Matches Node's
+/// `path.relative` / Python's `os.path.relpath`: shares the longest common
+/// prefix and emits `..` for each component in `from` past the common prefix.
+#[cfg(any(windows, test))]
+#[expect(
+    clippy::disallowed_types,
+    reason = "component-wise path arithmetic needs std Path/PathBuf; callers bridge to vite_path"
+)]
+fn relative_path(from: &std::path::Path, to: &std::path::Path) -> std::path::PathBuf {
```
Comment (wan9chi) [inline crates/vite_task_plan/src/ps1_shim.rs]: - Return type: Local returns PathBuf unconditionally; pathdiff returns Option<PathBuf> to signal unresolvable inputs.
- Absolute/relative mix: pathdiff short-circuits: returns Some(path) when path is absolute and base isn't, None otherwise. Local compares components blindly — a mixed input silently produces a garbage path (root component mismatches, so every from component becomes .. and to is appended whole).
- `..` in base: pathdiff returns None (can't resolve without filesystem). Local treats .. as a plain component and produces a wrong but non-failing result.
- `.` in base: pathdiff skips it (pushes a, keeps going). Local treats it as a real component, so a mismatch would break the common-prefix count.
- Allocation shape: Local materializes both sides into Vec<Component> then zips; pathdiff streams with iterators and only allocates the result vec.

This doesn't handle these edge cases well. We should use https://docs.rs/pathdiff/latest/pathdiff/

File: crates/vite_task_plan/tests/plan_snapshots/main.rs [hunk 3]
```diff
@@ -42,6 +42,12 @@ struct Plan {
 
 #[derive(serde::Deserialize, Default)]
 struct SnapshotsFile {
+    /// When `true`, the fixture only runs on Windows. Useful for cases whose
+    /// task-resolution or plan output depends on Windows-specific behavior
+    /// (e.g. PATHEXT-driven `.cmd` lookups), since those wouldn't resolve at
+    /// all under Unix-style `which::which_in`.
+    #[serde(default)]
+    pub windows_only: bool,
```
Comment (wan9chi) [inline crates/vite_task_plan/tests/plan_snapshots/main.rs]: Let's reuse the existing `platform: Option<Str>` field from the e2e harness ([e2e_snapshots/main.rs:193](https://github.com/voidzero-dev/vite-task/blob/892af621431cbc4219495784c14a95f9d260523b/crates/vite_task_bin/tests/e2e_snapshots/main.rs#L193)) instead of introducing `windows_only: bool`.

Also, let's filter the trial out during enumeration (like [e2e_snapshots/main.rs:514](https://github.com/voidzero-dev/vite-task/blob/892af621431cbc4219495784c14a95f9d260523b/crates/vite_task_bin/tests/e2e_snapshots/main.rs#L514)) rather than early-returning from the test body. Right now a skipped case reports as passed, which reads as "verified on this platform" when it wasn't actually run.

**Comment by fengmk2** [comment]
@cursor review
